src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3485,6 +3485,72 @@ private IOperation VisitNoneOperationExpression(IOperation operation)
             return Operation.CreateOperationNone(semanticModel: null, operation.Syntax, operation.ConstantValue, childrenBuilder.ToImmutableAndFree(), operation.IsImplicit);
         }
 
+        public override IOperation VisitInterpolatedString(IInterpolatedStringOperation operation, int? captureIdForResult)
+        {
+            foreach (IInterpolatedStringContentOperation element in operation.Parts)
+            {
+                if (element.Kind == OperationKind.Interpolation)
+                {
+                    var interpolation = (IInterpolationOperation)element;
+                    _evalStack.Push(Visit(interpolation.Expression));
+
+                    if (interpolation.Alignment != null)
+                    {
+                        _evalStack.Push(Visit(interpolation.Alignment));
+                    }
+
+                    if (interpolation.FormatString != null)
+                    {
+                        _evalStack.Push(Visit(interpolation.FormatString));
+                    }
+                }
+                else
+                {
+                    var interpolatedStringText = (IInterpolatedStringTextOperation)element;
+                    // PROTOTYPE(DATAFLOW): Should we avoid flow captures for constant string literal "Text" by not pushing onto the eval stack?
```

Comment:
>Should we avoid flow captures for constant string literal "Text" by not pushing onto the eval stack? [](start = 44, length = 100)

We don't do optimizations like this to decrease unnecessary complexity.

Reply:
However, in context of this specific node (IInterpolatedStringOperation) we probably shouldn't spill string literals under IInterpolatedStringTextOperation.

---
In reply to: [185615552](https://github.com/dotnet/roslyn/pull/26545#discussion_r185615552) [](ancestors = 185615552)

Reply:
Do you think we should make that change now or is it fine to tackle it later?

---
In reply to: [185616890](https://github.com/dotnet/roslyn/pull/26545#discussion_r185616890) [](ancestors = 185616890,185615552)

Reply:
I just checked the API [IInterpolatedStringTextOperation.Text](http://source.roslyn.io/#Microsoft.CodeAnalysis/Operations/IInterpolatedStringTextOperation.cs,22) has type `IOperation` instead of `ILiteralOperation`, though the current operation factory implementation always creates an `ILiteralOperation` for the text. I will add the literal text assert in the CFG builder and make the change in this PR.

---
In reply to: [185632011](https://github.com/dotnet/roslyn/pull/26545#discussion_r185632011) [](ancestors = 185632011)